omniscidb/QueryBuilder/QueryBuilder.h

```diff
@@ -0,0 +1,656 @@
+/**
+ * Copyright (C) 2022 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#include "IR/Exception.h"
+#include "IR/Node.h"
+#include "SchemaMgr/SchemaProvider.h"
+#include "Shared/Config.h"
+
+namespace hdk::ir {
+
+class InvalidQueryError : public Error {
+ public:
+  InvalidQueryError() {}
+  InvalidQueryError(std::string desc) : Error(std::move(desc)) {}
+};
+
+class QueryBuilder;
+class BuilderNode;
+class ExprRewriter;
```

Comment:
What is the rewriter exposed for?

Reply:
For `BuilderExpr::rewrite` interface

Reply:
What is the `rewrite` interface used for? I think I saw it for input replacement, but what are the major use cases?